examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/MCCommand.h

```diff
@@ -0,0 +1,39 @@
+/**
+ *
+ *    Copyright (c) 2020-2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#ifndef MCCommand_h
+#define MCCommand_h
+
+@interface MCCommand<RequestType, ResponseType> : NSObject
+
+/**
+ * @brief Invokes this MCCommand on the associated MCEndpoint and corresponding MCCluster
+ *
+ * @param request request data corresponding to this command invocation
+ * @param completion Called when command execution completes with nil NSError if successful and responseData. On failure, the NSError describes the error
+ * @param timedInvokeTimeoutMs command timeout in ms
```

Comment:
what does it mean to pass a nil timeout?

Reply:
A command invocation may done as a timed invocation or an untimed invocation (pass nil for timedInvokeTimeoutMs). See 

1.  [https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CHIPCluster.h#L99-L113](https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CHIPCluster.h#L99-L113)
2.  Matter spec section on "Invoke transaction": [https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/23a3e328bf29577628aab1eb1d69ca7d328aa761/src/data_model/Interaction-Model.adoc#invoke-interaction](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/23a3e328bf29577628aab1eb1d69ca7d328aa761/src/data_model/Interaction-Model.adoc#invoke-interaction)